### PR TITLE
doc: Add doc and script for ssh-demo sample

### PR DESCRIPTION
**Pull request by stevenhorsman**
- Add scripts and documentation to build, configure and test
the ssh-demo encrypted image sample in Kubernetes

Fixes: #3637

Signed-off-by: stevenhorsman <steven@uk.ibm.com>

### CODE REVIEW

File: docs/how-to/ccv0.sh [hunk 1]
```diff
@@ -260,14 +268,23 @@ create_a_local_rootfs() {
...
     pushd "${tests_repo_dir}"
     git checkout ${tests_branch}
     popd
     # During the ./rootfs.sh call the kata agent is built as root, so we need to update the permissions, so we can rebuild it
     sudo chown -R ${USER}:${USER} "${katacontainers_repo_dir}/src/agent/"
+
+    # If offline key broker set then include ssh-demo keys and config from
+    # https://github.com/confidential-containers/documentation/tree/main/demos/ssh-demo
+    if [ "${AA_KBC}" == "offline_fs_kbc" ]; then
+        sudo curl -Lo "${ROOTFS_DIR}/etc/aa-offline_fs_kbc-keys.json" https://raw.githubusercontent.com/confidential-containers/documentation/main/demos/ssh-demo/aa-offline_fs_kbc-keys.json
```
Comment (jodh-intel) [inline docs/how-to/ccv0.sh:283]: Please can you split this into two operations to avoid running `curl` as `root`?

File: docs/how-to/ccv0.sh [hunk 2]
```diff
@@ -78,31 +78,34 @@ Overview:
...
+- kubernetes_create_ssh_demo_pod:   Create a Kata CC runtime pod based on the ssh demo
+- connect_to_ssh_demo_pod:          Ssh into the ssh demo pod, showing that the decryption succeeded
+- kubernetes_delete_ssh_demo_pod:   Delete the Kata CC runtime pod based on the ssh demo
+- open_kata_console:                Stream the kata runtime's console
+- open_kata_shell:                  Open a shell into the kata runtime
+- agent_pull_image:                 Run PullImage command against the agent with agent-ctl
+- shim_pull_image:                  Run PullImage command against the shim with ctr
+- agent_create_container:           Run CreateContainer command against the agent with agent-ctl
+- test:                             Test using kata with containerd
+- test_capture_logs:                Test using kata with containerd and capture the logs in the user's home directory
```
Comment (jodh-intel) [inline docs/how-to/ccv0.sh:108]: Please could you sort this list alphabetically?

**Review by jodh-intel — COMMENTED** [review]
Thanks @stevenhorsman.

File: docs/how-to/ccv0.sh (hunk 2 above)
Comment (stevenhorsman) [reply docs/how-to/ccv0.sh]: I've added https://github.com/kata-containers/kata-containers/pull/3638/commits/f2243ffe34bb28a2d37b4c40afd68e1edada0dc8 in which I've sorted all the instructions. I'm happy to squash it into the main commit if you'd prefer?

**Comment by stevenhorsman** [comment]
Thanks very much for the review James, I hope I've made the changes you were after. Running the tests now :)

**Review by jodh-intel — APPROVED** [review]
Thanks @stevenhorsman.

**Pull request merged**